asterius/rts/rts.mblockalloc.mjs

```diff
@@ -30,31 +42,39 @@ export class MBlockAlloc {
   }
 
   allocMegaGroup(n) {
-    const req_blocks = ((rtsConstants.mblock_size * n) - rtsConstants.offset_first_block) / rtsConstants.block_size;
-    for (let i = 0; i < this.freeList.length; ++i) {
-      const bd = this.freeList[i],
-            blocks = this.memory.i32Load(bd + rtsConstants.offset_bdescr_blocks);
+    const req_blocks = 
+          Math.ceil(((rtsConstants.mblock_size * n) - rtsConstants.offset_first_block) / 
+          rtsConstants.block_size);
+
+    for (let i = 0; i < this.freeSegments.length; ++i) {
+      const [bd, r] = this.freeSegments[i];
```

Comment:
So, we are saving the (bd, right) tuple as a segment here, which in fact is not too far from the original design which only stores bd. Further more, the freeSegment function below still initializes the bdescr of freed space.

For simplicity, I think we should:

* Store (left, right) tuple, ensure left and right satisfy mblock alignment.
* freeSegment only takes care of wiping bytes.
* When allocMegaGroup picks up a tuple, it initializes the bdescr fields before returning.